esp-wifi/src/esp_now/mod.rs

```diff
@@ -349,6 +349,19 @@ impl<'d> EspNowManager<'d> {
         check_error!({ esp_now_add_peer(&raw_peer as *const _) })
     }
 
+    /// Set CSI configuration and registers the receiving callback.
+    pub fn set_csi(
+        &mut self,
+        mut csi: CsiConfig,
+        cb: fn(crate::binary::include::wifi_csi_info_t),
```

Comment:
We should probably take an `impl Fn(WifiCsiInfo)` here. We have the option to allocate it on heap, we don't need to constrain what the user can do.